### PR TITLE
Restructure folders

### PR DESCRIPTION
**Pull request by danitrod**
Opinionated example restructuring into separate categories (suggested in #401).

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,57 +1,58 @@
...
-   "websocket",
-   "websocket-autobahn",
-   "websocket-chat",
-   "websocket-chat-broker",
-   "websocket-tcp-chat"
+   "async/data_factory",
+   "async/db",
+   "async/ex1",
+   "async/ex2",
+   "async/pg",
```
Comment (robjtede) [inline Cargo.toml:7]: async/db -> database/basic (or database/sqlite)
async/pg -> database/pg
async/ex1 -> basics/json-validation
async/ex2 -> basics/nested-routing

**Review by robjtede — CHANGES_REQUESTED** [review]
Won't be too picky, in general this looks good.

File: Cargo.toml (hunk 1 above)
Comment (danitrod) [reply Cargo.toml:7]: What about async/data_factory? Move it to basics as well?

Comment (robjtede) [reply Cargo.toml:7]: basics or other, don't mind

**Comment by robjtede** [comment]
Some import fails to fix, if you would.

**Comment by danitrod** [comment]
> Some import fails to fix, if you would.

I'm not sure what you mean, do you mean the CI linting fails that are happening? Sorry for the confusion haha

**Comment by robjtede** [comment]
```
 Checking sqlx_todo v0.1.0 (/home/runner/work/examples/examples/database_interactions/sqlx_todo)
    Checking awc_https v0.1.0 (/home/runner/work/examples/examples/security/awc_https)
error: DATABASE_URL not set
Error:   --> database_interactions/sqlx_todo/src/todo/model.rs:43:20
   |
43 |           let recs = sqlx::query!(
   |  ____________________^
44 | |             r#"
45 | |                 SELECT id, description, done
46 | |                     FROM todos
47 | |                 ORDER BY id
48 | |             "#
49 | |         )
   | |_________^
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: DATABASE_URL not set
Error:   --> database_interactions/sqlx_todo/src/todo/model.rs:65:19
   |
65 |           let rec = sqlx::query!(
   |  ___________________^
66 | |                 r#"
67 | |                     SELECT * FROM todos WHERE id = $1
68 | |                 "#,
69 | |                 id
70 | |             )
   | |_____________^
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0433]: failed to resolve: use of undeclared crate or module `futures`
Error:  --> database_interactions/sqlx_todo/src/todo/model.rs:3:5
  |
3 | use futures::future::{ready, Ready};
  |     ^^^^^^^ use of undeclared crate or module `futures`

error[E0412]: cannot find type `Ready` in this scope
Error:   --> database_interactions/sqlx_todo/src/todo/model.rs:26:19
   |
26 |     type Future = Ready<Result<HttpResponse, Error>>;
   |                   ^^^^^ not found in this scope
   |
help: consider importing one of these items
   |
1  | use core::future::Ready;
   |
1  | use crate::env::sys::ext::net::raw_fd::sys_common::backtrace::backtrace_rs::symbolize::gimli::mystd::future::Ready;
   |
1  | use crate::web::payload::Ready;
   |
1  | use std::future::Ready;
   |

error[E0425]: cannot find function `ready` in this scope
Error:   --> database_interactions/sqlx_todo/src/todo/model.rs:31:9
   |
31 |         ready(Ok(
   |         ^^^^^ not found in this scope
   |
help: consider importing one of these items
   |
1  | use core::future::ready;
   |
1  | use crate::env::sys::ext::net::raw_fd::sys_common::backtrace::backtrace_rs::symbolize::gimli::mystd::future::ready;
   |
1  | use std::future::ready;
   |
```

**Comment by danitrod** [comment]
Hope I'm not bothering too much, but there's a specific error in the `sqlx_todo` example check about the `DATABASE_URL` env variable not being set. I checked in the github workflows but it doesn't seem to be setting any env variables, any directions on how I can fix this? Thanks for the fast answers and help btw

**Comment by robjtede** [comment]
yeah i'm not sure about that one, there's a different error there about ready() not being found... idk how that wasn't an error before though

**Comment by danitrod** [comment]
I managed to fix the other ones, a crate was actually missing from the example's `Cargo.toml`, maybe it wasn't failing before because of cache or something, but this one is weird to me, other database examples with the same env usage don't fail.

**Comment by robjtede** [comment]
ahh, seems that sqlx_todo was already omitted from the project list for this reason

**Comment by danitrod** [comment]
Ohh you're right, my bad. It actually wasn't in the workspace before and I added. gonna remove it for now then

**Comment by robjtede** [comment]
cool, we'll figure something out for it

**Review by robjtede — APPROVED** [review]

**Pull request merged**